server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -2980,4 +2980,21 @@ public void testSubscriptExpressionFromUnionAll() throws Exception {
         assertThat(analyzed.outputs()).hasSize(1);
         assertThat(analyzed.outputs().get(0)).isFunction("subscript", isField("obj"), isLiteral("unknown"));
     }
+
+    @Test
+    public void test_aliased_unknown_object_key() throws IOException {
+        var executor = SQLExecutor.builder(clusterService)
+            .addTable("create table t (o object)")
+            .build();
+        executor.getSessionSettings().setErrorOnUnknownObjectKey(false);
+        var analyzed = executor.analyze("select alias.o['unknown_key'] from (select * from t) alias");
+        assertThat(analyzed.outputs()).hasSize(1);
+        assertThat(analyzed.outputs().get(0)).isVoidReference("o['unknown_key']", new RelationName(null, "alias"));
+
+        // below is actually a bug but here to document and to prevent breaking changes
```

Comment:
We only throw an exception when it is a reference type:

https://github.com/crate/crate/blob/b1865f9362dab3e7eb707dc8a63b934c7e57a7e7/server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java#L710-L712

Reply:
Why? it's wrong behavior to not throw an error when the flag is `true`.

Reply:
I believe this behaviour was like this even before `error_on_unknown_object_key`(true by default) was introduced. There was a short [discussion](https://github.com/crate/crate/pull/11635#discussion_r691262154) and moved forward with keeping the behaviour. This can be re-discussed but would be good as separately. If so, I would take this part of the test out first.

Reply:
Let's hear other's opinion, but I believe when you set this to `true` (by default), we should throw an error regardless of aliased or un-aliased relations.
throwing an error for:
```
select o['unknown_key'] from t;
```
but not for:
```
select o['unknown_key'] from t as tt;
```
is a bug imho, and not a breaking change. If fixing this affects other scenarios, we should fix the other scenarios.

Reply:
Yes, this should raise an error.

Reply:
Is this now addressed?

Reply:
Applying a fix for this opened up a hidden bug : `test_can_access_element_within_object_array_of_derived_table_containing_a_union`. During analysis, sub-columns of objects of aliased tables cannot be matched anymore and treated as unknown causing `ColumnUnknownException`. The test is only for a union but other types need to be check as well.